src/bacnet/basic/object/netport.c

```diff
@@ -3036,6 +2820,75 @@ bool Network_Port_IPv6_Gateway_Zone_Index_Set(
 }
 
 #if (BBMD_CLIENT_ENABLED) && defined(BACAPP_HOST_N_PORT)
+#define MAX_LABEL_LENGTH 63
+static bool is_valid_hostname(const BACNET_CHARACTER_STRING *const hostname)
```

Comment:
please move is_valid_hostname() into either hostnport.c or into bacstr.c (similar to utf8 check) and prefix appropriately.

Reply:
Will do.

Reply:
done.